features/storage/nvstore/source/nvstore.cpp

```diff
@@ -579,6 +611,24 @@ int NVStore::garbage_collection(uint16_t key, uint16_t flags, uint8_t owner, uin
     return ret;
 }
 
+int NVStore::set_offset_by_key()
```

Comment:
This code is better as part of the `set_max_keys` function and not as a separate function. This way, you won't need `_old_max_keys`.